invoiceless payment plugin

gallizoltan opened the pull request:
- sends payment without an invoice from the receiving node
- uses circular payments: takes the money to the target node, pays in the form of routing fee, and brings some change back
- no plugin/update required on the receiving side

m-schmoock commented:
+1 for the funny idea :D
I have not gone through the code yet or tried...

m-schmoock commented on sendinvoiceless/sendinvoiceless.py:
```diff
@@ -0,0 +1,119 @@
+            route = forth['route'] + back['route']
+            setup_routing_fees(plugin, route, change, payload)
+            fees = route[0]['msatoshi'] - route[-1]['msatoshi'] - msatoshi
+            if fees > exemptfee and fees > msatoshi * float(maxfeepercent) / 100:
```
This will (currently) fail due to unresolved https://github.com/ElementsProject/lightning/pull/2601

I suggest the following workaround (as included in my improve rebalance PR):
```Python
if fees > exemptfee and int(fees) > int(msatoshi) * float(maxfeepercent) / 100:
```

gallizoltan replied:
Nice catch! I'm already waiting for the next pylightning release. Currently it causes exception if someone gives the rebalance/sendinvoiceless `msatoshi` in the form of "1sat". https://github.com/ElementsProject/lightning/pull/2581

m-schmoock commented:
No need to wait. Just use the source from the daemon `PYTHONPATH=./contributed/pylightning love lightningd --daemon --plugin=...`

m-schmoock commented:
BTW, christian merged my fix already

cdecker commented:
Very nice plugin. Could you maybe add a short description and link in the table in the top-level `README.md` as well?

gallizoltan commented:
@cdecker Good idea! I've just added description and link in the top-level `README.md`. And rebased to the tip of lightningd/plugins to not get a conflict at merging.
